tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -1732,8 +1732,10 @@ private[optimizer] abstract class OptimizerCore(
   private def transformExprsOrSpreads(trees: List[Tree])(
       implicit scope: Scope): List[Tree] = {
 
-    trees match {
-      case (spread: JSSpread) :: rest =>
+    val builder = List.newBuilder[Tree]
+
+    trees.foreach {
```

Comment:
It is, but `flatMap` will create a lot of garbage lists in this case, since the `JSSpread` case should be fairly rare.

In general we don't have any `flatMap` on collections in `OptimizerCore.scala` (we have one `flatMap`s on `Option`s).